### ISSUE TITLE
Feature request: A more compact view?

### ISSUE THREAD

**Issue by Gruntfuggly** [DonJayamanne/gitHistoryVSCode#90]
Would it be possible to add an option to show the details of each commit all on one line? Or to reduce the padding around each commit entry?

**Comment by mikes-gh** [comment]
Just so I understand this more is this so you do less scrolling or you just think there's too much space or you think it doesn't look right.
Are you on a smaller screen?

Understanding the reason would help asses this.
Thanks in anticipation of you feeback

**Comment by Gruntfuggly** [comment]
I'm used to using gitk, which displays the graph more compactly.

I guess it's because I like to be able to see more context. When you're working with several branches, it's useful to be able to see a lot more of the tree at the same time.

I'm probably also asking in anticipation of the graph being able to show all branches 8-)

**Comment by mikes-gh** [comment]
Understood.  Seems like a good argument.  Probably be an option unless everyone is of the same opinion.
Comments welcome.

**Comment by jshall** [comment]
I would prefer a one-line layout, but it definitely needs to take #45 into account.
It would be even better if users could customize the layout via angular-style templates and/or CSS.

**Comment by nonameolsson** [comment]
I think this sounds good. A more compact view would be much better!

Something like this is what I would like to see:
![webstorm](https://d3nmt5vlzunoa1.cloudfront.net/idea/files/2016/09/ezgif.com-optimize-2.gif)

This is the solution from Atom [https://atom.io/packages/git-log](https://atom.io/packages/git-log)
![atom](https://i.github-camo.com/fe46952e7b204bcc30575e592b23bd791e40b25d/68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f4e696b68696c4b616c6967652f6769742d6c6f672f6d61737465722f7265736f75726365732f6769742d6c6f672e676966)
![atom2](https://cdn-business.discourse.org/uploads/github_atom/24534/7a8abfac55022b08.png)

What do you think @Gruntfuggly ?

**Comment by DonJayamanne** [comment]
@nonameolsson 
I like the first option (screen sample). What application is that from?

**Comment by nonameolsson** [comment]
@DonJayamanne That is from JetBrains PhpStorm. 
https://www.jetbrains.com/help/phpstorm/2016.3/log-tab.html

But it is the same in all their products. You could download WebStorm 30 day trial to try it out yourself [https://www.jetbrains.com/webstorm/](https://www.jetbrains.com/webstorm/)

**Comment by Gruntfuggly** [comment]
Yes, something similar is what I would like to be able to see. I like the atom view - nice and compact 8-)

**Comment by jshall** [comment]
I prefer the layout shown for Atom, but love the array of options shown for JetBrains.

**Comment by nonameolsson** [comment]
Will you try to implement something like that?

**Comment by DonJayamanne** [comment]
I will start on some of this work

**Comment by DavidHenri008** [comment]
Any updates on this improvement?
Thanks.

**Comment by DonJayamanne** [comment]
@ole1986 I'm going to work on this, I think we need a oneline view.

**Comment by nonameolsson** [comment]
Is this still under development? :)
